rusty_pizza_server/src/money.rs

```diff
@@ -0,0 +1,383 @@
+    }
+
+    #[test]
+    fn u8_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u8 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u16() {
+        // When:
+        let result = Money::new(5, 0) * 2u16;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u16_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u16;
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn u16_can_be_multiplied_with_money() {
+        // When:
+        let result = 2u16 * Money::new(5, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn u16_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u16 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u32() {
+        // When:
+        let result = Money::new(5, 0) * 2u32;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u32_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u32;
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn u32_can_be_multiplied_with_money() {
+        // When:
+        let result = 2u32 * Money::new(5, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn u32_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u32 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_is_copy() {
+        // Given:
+        let money = Money::new(2, 0);
+
+        // When:
+        let result1 = 3u32 * money;
+        let result2 = money + Money::new(1, 0);
+
+        // Then:
+        assert_eq!(result1, Money { cents: 600 });
+        assert_eq!(result2, Money { cents: 300 });
+    }
+
+    #[test]
+    fn money_prints_as_euro_amount() {
+        // Given:
+        let money = Money::new(2, 99);
+
+        // When:
+        let mut output = String::new();
+        write!(&mut output, "{}", money).expect("Error formatting money");
+
+        // Then:
+        assert_eq!(output, "2,99€");
+    }
+
+    #[test]
+    fn can_get_euros_from_money() {
+        // Given:
+        let money = Money::new(2, 99);
+
+        // When:
+        let result = money.get_euros();
+
+        // Then:
+        assert_eq!(result, 2);
+    }
+
+    #[test]
+    fn can_get_euros_from_money_with_alternative_values() {
+        // Given:
+        let money = Money::new(3, 99);
```

Comment:
For alternative value please change the cent value, too.

Reply:
Done.